pkg/executor/sortexec/sort.go

```diff
@@ -124,11 +131,13 @@ func (e *SortExec) Close() error {
 		// will use `e.Parallel.workers` and `e.Parallel.merger`.
 		channel.Clear(e.Parallel.resultChannel)
 		for i := range e.Parallel.workers {
-			e.Parallel.workers[i].batchRows = nil
-			e.Parallel.workers[i].localSortedRows = nil
-			e.Parallel.workers[i].sortedRowsIter = nil
-			e.Parallel.workers[i].merger = nil
-			e.Parallel.workers[i].memTracker.ReplaceBytesUsed(0)
+			if e.Parallel.workers[i] != nil {
+				e.Parallel.workers[i].batchRows = nil
+				e.Parallel.workers[i].localSortedRows = nil
+				e.Parallel.workers[i].sortedRowsIter = nil
+				e.Parallel.workers[i].merger = nil
+				e.Parallel.workers[i].memTracker.ReplaceBytesUsed(0)
```

Comment:
maybe you can wrap L135-L139 into a function like `reset/close/clear`

Reply:
> maybe you can wrap L135-L139 into a function like `reset/close/clear`

done